### ISSUE TITLE
Consider supporting column-aligned indentation levels when wrapping binary expressions

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#32830]
**Version Used**: Master branch once #32809 is merged

**Steps to Reproduce**:

Wrap the binary expression in the following code:

```csharp
while (a && b)
{
}
```

**Expected Behavior**:

Users should have an option to wrap the expression with wrapped sub-expressions aligned to the original expression, like this:

```csharp
while (a &&
       b)
{
}
```

**Actual Behavior**:

Users only have the option to wrap the expression the way Smart Indent would wrap the expression when pressing <kbd>Enter</kbd>.

```csharp
while (a &&
    b)
{
}
```

**Comment by CyrusNajmabadi** [comment]
Note: this behavior is already in the product.  The only reason it would go away would be if it was unnecessarily removed.

**Comment by alrz** [comment]
>Removing support for this goes against the entire point of the refactoring in the first place.

I also think "alignment" and "wrapping" are two separate formatting concerns. You may want to align lines but not wrap other operators. So the refactoring would only do the "wrapping" part and alignment would be taken out for another option. Does that make sense?

**Comment by CyrusNajmabadi** [comment]
could you clarify what you mean by "another option"?  Do you mean an actual user option?  Or something else?  I'm opposed to this being controlled by an option that controls how things work everywhere.

One thing i've learned about thsi space is that a single codebase/user will often format these wrapped constructs differently in different contexts.  it's all about what looks sensible in a given location.  So whatever solution we have, i'd want to be able to make sure that people have quick access to the precise formatting they want right there.

--

Note: having wrapping/alignment pulled out into differnet refactorings is a potential design.  I don't particular like it though because of the need to now invoke things multiple times.  The arg/param wrapping code is nice in that:

1. it gives quick access to the top format/wrapping options people use.
2. it learns what you like, so that it can provide those options near the top, making it easy to get to them with minimal effort.

I very much like where we landed there and it's unclear to me why we couldn't land there as well for the other wrapping cases.

**Comment by sharwell** [comment]
Design review outcome: We should provide both options, similar to how Wrap Parameters provides both options.

**Comment by jinujoseph** [comment]
closed via #32859
